### ISSUE TITLE
Circuit connecttion

### ISSUE THREAD

**Issue by ManasKumarPanigrahi** [RyanZotti/Self-Driving-Car#89]
Hi Ryan, how can I get to know about the circuit connection?

**Comment by RyanZotti** [comment]
@ManasKumarPanigrahi Yes, this is a common request I've gotten. I definitely need to document how I did the wiring. I plan to have the wiring documentation finished in 1-2 weeks. I'll reply back here once complete.

**Comment by ManasKumarPanigrahi** [comment]
thank u. waiting for your reply.

**Comment by RyanZotti** [comment]
@ManasKumarPanigrahi I'm about halfway through the wiring docs. It's taking awhile because there are a LOT of wires, and at the same time I want to be sure I make my docs accurately. If you need something in the interim, these are the two sites I originally used: [motor](https://business.tutsplus.com/tutorials/controlling-dc-motors-using-python-with-a-raspberry-pi--cms-20051), and [range sensor](https://www.modmypi.com/blog/hc-sr04-ultrasonic-range-sensor-on-the-raspberry-pi). The motor documentation is for the the Pi 2, so I had to make some modifications. Anyways, I still plan to reply back here once the docs are 100% complete. I'll also make them part of the main README.md.

**Comment by RyanZotti** [comment]
@ManasKumarPanigrahi I've completed the circuitry diagram. You can find a magnified example [here](https://user-images.githubusercontent.com/8901244/30244462-0e34e4be-9573-11e7-8ea8-81f1203c9492.png). It's also now on the main page/readme. Let me know if you have any questions.

**Comment by prateekralhan** [comment]
@RyanZotti Firstly, Your project is really superb!! Kudos to you!! I had just one doubt that can you please help me with this thing... that how and where did you interface your camera with your Pi.

**Comment by RyanZotti** [comment]
@prateekralhan I used a USB webcam, so I just plugged it into one of the Pi's 4 USB ports.

**Comment by prateekralhan** [comment]
@RyanZotti Sir, Thanks for the Info!! You seem to be using a Logitech webcam, Can I interface another camera or even the raspberry camera?? Will the same codes work for It? Or Do I need to Make any changes in its code!

**Comment by RyanZotti** [comment]
@prateekralhan Yes, you can use the Raspberry Pi camera. The software for the Pi camera is much simpler and easier to use, however it is much less durable. My car has many exposed wires, and I kept accidentally destroying my Pi cameras. The Logitech webcam is cheap and durable -- its wires are protected by a thick layer of plastic unlike the Pi camera. The Logitech cable is also longer, so I was able to house the Pi inside the car and mount the camera on the car's roof. The Pi camera's wiring wasn't long enough to do this, so I would have had to put the Pi and the camera on the car's roof, and that would have looked a lot uglier. 

If you don't care about aesthetics and want something that is easier to install (ffmpeg is very difficult to install), go with the Pi camera and mount both the Pi and its camera on top of the car.

**Comment by prateekralhan** [comment]
@RyanZotti Thank You So Much Sir!! I am Doing a bit different!! I am trying to setup my smartphone's camera for now!! As I have already spend a lot!! Just trying to create an app that sort of acts as both a TCP server,receiving streamed image frames from a video streaming app on a phone and
a user interface allowing a human driver to control the car with the cursor keys or mouse, sent to an RC unit over the serial interface. Don't Know whether it shall work!! Any ideas regarding this Sir!!??

**Comment by RyanZotti** [comment]
@ManasKumarPanigrahi I haven't written code for a phone before, but if it runs Linux I would just use Python's `tornado` package for the TCP server like I did with the Pi. I'm not sure how you would stream the video, though I guess it's possible you could use ffmpeg for that as well. It's definitely possible to use a phone -- I've seen others successfully use a phone.
